### ISSUE TITLE
overload declaration for arbitrary-length tuple parameter

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#6654]
**Describe the bug**

overload declaration takes fixed sized tuple for function takes arbitrary length tuple causes error.

**Code or Screenshots**

```python
from typing import Tuple, overload

@overload
def f(t: Tuple[()]) -> int:
    ...

@overload
def f(t: Tuple[int]) -> int:
    ...

@overload
def f(t: Tuple[int, int]) -> int:
    ...

@overload
def f(t: Tuple[int, int, int]) -> int:
    ...

def f(t: Tuple[int, ...]) -> int:
    return 0
```

```
$ pyright ol.py 
ol.py
  ol.py:19:5 - error: Overloaded implementation is not consistent with signature of overload 1
    Type "(t: Tuple[int, ...]) -> int" cannot be assigned to type "(t: Tuple[()]) -> int"
      Parameter 1: type "Tuple[()]" cannot be assigned to type "Tuple[int, ...]"
        "Tuple[()]" is incompatible with "Tuple[int, ...]"
          Tuple size mismatch; expected 1 but received 0 (reportGeneralTypeIssues)
  ol.py:19:5 - error: Overloaded implementation is not consistent with signature of overload 2
    Type "(t: Tuple[int, ...]) -> int" cannot be assigned to type "(t: Tuple[int]) -> int"
      Parameter 1: type "Tuple[int]" cannot be assigned to type "Tuple[int, ...]"
        "Tuple[int]" is incompatible with "Tuple[int, ...]"
          Tuple size mismatch; expected 1 but received 1 (reportGeneralTypeIssues)
  ol.py:19:5 - error: Overloaded implementation is not consistent with signature of overload 3
    Type "(t: Tuple[int, ...]) -> int" cannot be assigned to type "(t: Tuple[int, int]) -> int"
      Parameter 1: type "Tuple[int, int]" cannot be assigned to type "Tuple[int, ...]"
        "Tuple[int, int]" is incompatible with "Tuple[int, ...]"
          Tuple size mismatch; expected 1 but received 2 (reportGeneralTypeIssues)
  ol.py:19:5 - error: Overloaded implementation is not consistent with signature of overload 4
    Type "(t: Tuple[int, ...]) -> int" cannot be assigned to type "(t: Tuple[int, int, int]) -> int"
      Parameter 1: type "Tuple[int, int, int]" cannot be assigned to type "Tuple[int, ...]"
        "Tuple[int, int, int]" is incompatible with "Tuple[int, ...]"
          Tuple size mismatch; expected 1 but received 3 (reportGeneralTypeIssues)
4 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**

command-line tool, 1.1.338

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.
